src/shell/output.c

```diff
@@ -204,6 +207,10 @@ static int shell_output_commit (struct shell_output *out)
         out->output_ready_sent = true;
     if (flux_future_then (f, -1, shell_output_commit_completion, out) < 0)
         goto error;
+    if (flux_shell_add_completion_ref (out->shell, "output.commit") < 0) {
```

Comment:
With the completion references taken for outstanding futures, will `shell_output_destroy()` ever be called with a non-empty `out->pending_writes`? (Maybe tracking futures in the list is not necessary anymore?)

Reply:
I actually had pondered that, and thought there were still circumstances it was needed, such as if the shell was signaled.  But now that I look at it again, perhaps its not needed afterall.  I'll remove.